browser/tor/tor_profile_service.cc

```diff
@@ -56,15 +56,63 @@ void TorProfileService::RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
   const std::string tor_proxy_uri =
       std::string(kTorProxyScheme) + std::string(kTorProxyAddress) + ":" + port;
   registry->RegisterStringPref(prefs::kTorProxyString, tor_proxy_uri);
-  // Tor for group policy. Only can be true by group policy.
+  // kTorDisabled is managed prefs.
+  // User can only change it via settiongs option if it's not managed.
   registry->RegisterBooleanPref(prefs::kTorDisabled, false);
+  // kTorDisabledAtNextLaunching has the value that user changed via settings.
+  // If this value is changed, it is exposed to settings page. At the next
+  // launching, this value is copied to kTorDisabled.
+  registry->RegisterBooleanPref(prefs::kTorDisabledAtNextLaunching, false);
+}
+
+// static
+void TorProfileService::InitializeTorPrefs() {
+  auto* state = g_browser_process->local_state();
+  if (state->FindPreference(
+          prefs::kTorDisabledAtNextLaunching)->IsDefaultValue()) {
+    return;
+  }
+
+  state->SetBoolean(prefs::kTorDisabled,
+                    state->GetBoolean(prefs::kTorDisabledAtNextLaunching));
+  state->ClearPref(prefs::kTorDisabledAtNextLaunching);
 }
 
 // static
 bool TorProfileService::IsTorDisabled() {
   return g_browser_process->local_state()->GetBoolean(prefs::kTorDisabled);
 }
 
+// static
+bool TorProfileService::IsTorDisabledAtNextLaunching() {
+  return g_browser_process->local_state()->GetBoolean(
+      prefs::kTorDisabledAtNextLaunching);
+}
+
+// static
+bool TorProfileService::IsTorDisabledManaged() {
```

Comment:
`IsTorDisabledManaged` is even more confusing ;) `IsTorManaged` would be the convention and the js methods should match `getTorManaged` -> `isTorManaged`. It's much easier to understand things when we don't arbitrarily use different names for the same thing in different places

Reply:
Yeah, this looks confusing. So, I deleted this method because `BraveDefaultExtensionsHandler::IsTorManaged()` is the only caller.